tests-e2e/ocw-ci-test-course/resource-list.spec.ts

```diff
@@ -0,0 +1,36 @@
+import { test, expect } from "@playwright/test"
+import { CoursePage } from "../util"
+
+test("Resource list shows correct thumbnails and aria labels for different types of files", async ({
+  page
+}) => {
+  const course = new CoursePage(page, "course")
+  await course.goto("/lists/a-resource-list")
+  const resourceLocators = await page
+    .locator(".resource-list-item > .row")
+    .all()
+
+  for (const resource of resourceLocators.values()) {
```

Comment:
**Requested change:** Don't loop over a dynamic list here for the reasons below.

Does anything guarantee that this list is not empty? I.e., I think this test would pass if we changed the CSS from `.resource-list-item` to `.resource-list li` in the templates. The list would be empty, and no assertions would fail.

In general, I avoid looping over dynamic lists in tests. And if I need to, I first make an assertion about its length to avoid the above scenario.

**But:** IMO, a better approach here would be to NOT loop over a dynamic list. Instead, pick a definite list of resources you're going to check:

```ts
  const expectedResoureces = [
    {
      title:     "file.mp4",
      thumbnail: {
        src:       "/static_shared/images/mobile_video_thumbnail.png",
        ariaLabel: "Video File"
      }
    },
    // add more
  ]

  for (const expected of expectedResoureces) {
    // ideally this would be page.getByRole("item"), but our HTML is not very semantic
    const resource = await page.locator(".resource-list-item", {
      has: page.getByRole("link", { name: expected.title })
    })

    const thumbnail = await resource.getByRole("img", { name: "Video File" })
    await expect(thumbnail).toBeVisible()
    const src = await thumbnail.getAttribute("src")
    expect(src).toBe(expected.thumbnail.src)
  }
```
because:
- now you're confident on the assertions your making
- you're sure you're checking different thumbnails
- much more explicit... As written, it's not clear whether this test is checking mp4 files or not. (It is, since the resource mock exists, but if someone removed the mock, the test would pass, and it would not be.)